util/reggen/multi_register.py

```diff
@@ -169,3 +207,39 @@ def _asdict(self) -> Dict[str, object]:
         rd['compact'] = str(self.compact)
 
         return {'multireg': rd}
+
+    def apply_alias(self, alias_reg: object, where: str) -> None:
+        '''
+        Compare all attributes and replace overridable values.
+
+        This ensures both registers are identical up to the overridable
+        attributes like 'name', 'desc', 'resval' and 'tags'.
+        '''
+        if not isinstance(alias_reg, MultiRegister):
+            raise ValueError('alias_reg is not of type MultiRegister')
+
+        # First, validate and update the multireg itself.
+        for key, val in self.__dict__.items():
+            # This is an overridable attribute, so replace the value.
+            if key in ALIAS_REG_ATTRS:
+                setattr(self, key, getattr(alias_reg, key))
+            # This is not an overridable attribute, so ensure values match.
+            elif key in NON_ALIAS_REG_ATTRS:
+                if getattr(alias_reg, key) != val:
+                    raise ValueError('Value mismatch for attribute {} '
+                                     'between alias multiregister {} and '
+                                     'multiregister {} ({} != {}) '
+                                     'in {}.'
+                                     .format(key,
+                                             alias_reg.name,
+                                             self.name,
+                                             getattr(alias_reg, key),
+                                             val,
+                                             where))
+
+        # Then, update the template register.
+        self.reg.apply_alias(alias_reg.reg, where)
+
+        # Finally, iterate over expanded regs and update them as well.
+        for creg, alias_creg in zip(self.regs, alias_reg.regs):
```

Comment:
Do we know that these are the same length? I can believe that we might have checked it (and, if not, we probably should). But could we also add an assertion?

Reply:
Since the multiregs must have the same structure, this must match at this point. 
But you make a good point about adding an assertion - let me do that.